hooks/useLogin.ts

```diff
@@ -54,12 +54,13 @@ export function useLogin(config: LoginProps = {}) {
         } catch (error) {}
 
         if (!token) {
-          if (redirect && typeof redirect === 'string') {
+          if (redirect) {
```

Comment:
redirect is a string, so if its truthy it is already proven to TSC that it is a string